src/promptflow-devkit/promptflow/_sdk/schemas/_flow.py

```diff
@@ -71,32 +71,41 @@ class FlowSchema(BaseFlowSchema):
     ValueType.OBJECT.value,
 ]
 
+ALLOWED_INIT_PARAM_TYPES = [
+    InitParamType.AZURE_OPEN_API_MODEL_CONFIGURATION.value,
+    InitParamType.OPEN_AI_MODEL_CONFIGURATION.value,
+    InitParamType.AZURE_OPEN_AI_CONNECTION.value,
+    InitParamType.OPEN_AI_CONNECTION.value,
+    InitParamType.QDRANT_CONNECTION.value,
+    InitParamType.COGNITIVE_SEARCH_CONNECTION.value,
+    InitParamType.SERP_CONNECTION.value,
+    InitParamType.AZURE_CONTENT_SAFETY_CONNECTION.value,
+    InitParamType.FORM_RECOGNIZER_CONNECTION.value,
+    InitParamType.WEAVIATE_CONNECTION.value,
+    InitParamType.SERVERLESS_CONNECTION.value,
+    InitParamType.CUSTOM_CONNECTION.value,
+]
+
 
 class FlexFlowInputSchema(FlowInputSchema):
     type = fields.Str(
         required=True,
         # TODO 3062609: Flex flow GPT-V support
-        validate=validate.OneOf(ALLOWED_TYPES),
+        validate=validate.OneOf(ALLOWED_PRIMITIVE_TYPES),
     )
 
 
 class FlexFlowInitSchema(FlowInputSchema):
     type = fields.Str(
         required=True,
-        validate=validate.OneOf(
-            ALLOWED_TYPES
-            + list(
-                map(lambda x: f"{x.value}Connection", filter(lambda x: x != ConnectionType._NOT_SET, ConnectionType))
-            )
-            + list(MODEL_CONFIG_NAME_2_CLASS.keys())
-        ),
+        validate=validate.OneOf(ALLOWED_PRIMITIVE_TYPES + ALLOWED_INIT_PARAM_TYPES),
```

Comment:
do we support object type?

Reply:
here "primitive" is confusing. object is already included in ALLOWED_PRIMITIVE_TYPES